### ISSUE TITLE
Update status with error  when a new version of collection is not activated due to error condition

### ISSUE THREAD

**Issue by mtamboli** [kabanero-io/kabanero-operator#311]
I updated Kabanero CR to a new release of collection. It seems like that one of the collections has some errors in the customization of pipelines. So all of the collections were activated to new release except the one which had issues. This behavior is correct. But it would  be nice to display the error in collection status or some other place which indicates there is a problem with new collections.

**Collection which had issue activating new version:**
```
spec:
  desiredState: active
  name: java-microprofile
  repositoryUrl: >-
    https://github.com/mtamboli/collections/releases/download/v0.3.6/kabanero-index.yaml
  version: 0.2.19
status:
  activeLocation: >-
    https://github.com/mtamboli/collections/releases/download/v0.3.5/kabanero-index.yaml
  activePipelines:
```
**Collection which got activated to the new version correctly:**
```
spec:
  desiredState: active
  name: java-spring-boot2
  repositoryUrl: >-
    https://github.com/mtamboli/collections/releases/download/v0.3.6/kabanero-index.yaml
  version: 0.3.16
status:
  activeLocation: >-
    https://github.com/mtamboli/collections/releases/download/v0.3.6/kabanero-index.yaml
  activePipelines:
```
Had to look at the operator log to know there was an error:
```
Header names:  ./,./build-push-deploy-pipeline.yaml,./._deploy-task.yaml,./deploy-task.yaml,./image-retag-push-task.yaml,./manifest.yaml,./._test-task.yaml,./test-task.yaml,./._image-scan-task.yaml,./image-scan-task.yaml,./._test-build-push-deploy-pipeline.yaml,./test-build-push-deploy-pipeline.yaml,./._build-push-task.yaml,./build-push-task.yaml
{"level":"error","ts":1574274128.83154,"logger":"kubebuilder.controller","msg":"Reconciler error","controller":"collection-controller","request":"kabanero/java-microprofile","error":"File ./._deploy-task.yaml was found in the archive, but not in the manifest.yaml","stacktrace":"github.com/kabanero-io/kabanero-operator/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\tsrc/github.com/kabanero-io/kabanero-operator/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/kabanero-io/kabanero-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsrc/github.com/kabanero-io/kabanero-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:217\ngithub.com/kabanero-io/kabanero-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1\n\tsrc/github.com/kabanero-io/kabanero-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158\ngithub.com/kabanero-io/kabanero-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\tsrc/github.com/kabanero-io/kabanero-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\ngithub.com/kabanero-io/kabanero-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil\n\tsrc/github.com/kabanero-io/kabanero-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134\ngithub.com/kabanero-io/kabanero-operator/vendor/k8s.io/apimachinery/pkg/util/wait.Until\n\tsrc/github.com/kabanero-io/kabanero-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
```

**Comment by mtamboli** [comment]
I configured CLI to see the status update in this case.

@davco01a  I think this is not correct. It should somehow show that the latest version in Git is not active. Problem could be because version of collection hub is updated but actual collection is same? 
```
./kabanero list

Kabanero Instance Collections	Version	Status
-----------------------------	-------	------
java-ee				0.1.1	active
java-microprofile		0.2.19	active
java-spring-boot2		0.3.16	active
nodejs				0.2.6	active
nodejs-express			0.2.8	active
nodejs-loopback			0.1.6	active

GitHub Curated Collections	Version
--------------------------	-------
java-ee				0.1.1
java-microprofile		0.2.19
java-spring-boot2		0.3.16
nodejs				0.2.6
nodejs-express			0.2.8
nodejs-loopback			0.1.6
```
./kabanero sync
kabanero instance collections is already synchronized with the github curated collections

**Comment by kaczyns** [comment]
From the operator perspective, the issue here is that there are files like `._deploy-task.yaml` in one of the pipeline zips in the java-microprofile collection (note the leading dot and underscore characters).  These files do not have entries in the `manifest.yaml` (should have a SHA256 in there), and also, they are not valid yaml files, they look like binary data.  The operator should somehow put the error from the log (`"File ./._deploy-task.yaml was found in the archive, but not in the manifest.yaml"`) into the collection status, so the user can see what is wrong without going to the kabanero-operator log.

A second issue should be opened to resolve the CLI output.  I am not sure if the collection version is the same (old/new), but it would be nice if the CLI could show somehow that there is a problem.  We may need to brainstorm about how to do this - from the operator's perspective, there is a valid version of the java-microprofile collection active... and therefore the CLI may be showing the right thing.  But then again, there is a (newer?) version of the collection that is not active yet, and will probably replace the currently active version.

**Comment by mtamboli** [comment]
Opened CLI issue: https://github.com/kabanero-io/kabanero-command-line/issues/141

> A second issue should be opened to resolve the CLI output. I am not sure if the collection version is the same (old/new), but it would be nice if the CLI could show somehow that there is a problem. We may need to brainstorm about how to do this - from the operator's perspective, there is a valid version of the java-microprofile collection active... and therefore the CLI may be showing the right thing. But then again, there is a (newer?) version of the collection that is not active yet, and will probably replace the currently active version.

**Comment by davco01a** [comment]
updated kabanero-io/kabanero-command-line#141

**Comment by davco01a** [comment]
The collection Object in the example at the top for the one that has an issue does not show the subfield `status` under Status, e.g.:

should be:

```
status:
  activeLocation: >-
    https://github.com/mtamboli/collections/releases/download/v0.3.5/kabanero-index.yaml
  activePipelines:
  status: inactive
```

**Comment by davco01a** [comment]
RE: 
``
@davco01a I think this is not correct. It should somehow show that the latest version in Git is not active. Problem could be because version of collection hub is updated but actual collection is same?
```

The CLI Service basis whether something is active or not based on `status->status` if the operator hasn't made that current we cannot reflect it in the CLI

**Comment by davco01a** [comment]
Please try again at 0.4.0

**Comment by marikaj123** [comment]
Fixing the problem with the error message not displaying on the Kabanero CR instance for v.0.5.0.

**Comment by kaczyns** [comment]
The operator is fixed in master and should be included in Kabanero release 0.5.0.  The CLI issue is still open.  I am closing the operator issue (this issue).
